segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -283,10 +282,12 @@ protected void doStart() {
                     if (ex == null) {
                         // Successful start.
                         log.info("{}: Started.", this.traceObjectId);
-                    } else if (!(Exceptions.unwrap(ex) instanceof ObjectClosedException) || !Services.isTerminating(state())) {
+                    } else if (Services.isTerminating(state())) {
                         // If the delayed start fails, immediately shut down the Segment Container with the appropriate
-                        // exception. We should ignore ObjectClosedExceptions or other exceptions during a shutdown phase
-                        // since that's most likely due to us shutting down.
+                        // exception. However if we are already shut down (or in the process of), it is sufficient to
```

Comment:
"already shut down" is the correct representation ("to be shut down" == in a state where we are not operating).
"in the process of" is indeed missing "shutting down" but it can be inferred from the rest of the sentence. 

This comment indicates that we can either be totally shut down or be in the process of shutting down (they are different states when it concerns a Service).